backend/server.ts

```
@@ -0,0 +1,366 @@
+  project?: string;
+  [key: string]: any;
+}
+
+app.post("/confirm-costs", 
+  strictLimiter,
+  body('data').isArray().withMessage('Data must be an array'),
+  body('data.*.id').notEmpty().withMessage('Each element must have an id'),
+  body('project').optional().trim().notEmpty(),
+  handleValidationErrors,
+  haltOnTimedout,
+  async (req: Request<{}, {}, KafkaMessageBody>, res: Response) => {
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)